### PR TITLE
Fix use string pattern on bytes-like object

### PR DESCRIPTION
**Pull request by chess99**
solved:
1. deadlock on proc.poll( )
2. TypeError: cannot use a string pattern on a bytes-like object

### CODE REVIEW

**Comment by pmeenan** [comment]
It would be better to pass universal_newlines=True to the Popen instead.  That way it will be text output on both Pythoin 2.7 and 3.x.  Otherwise, I think the decode might fail on Python 2.7 (which the code still supports).

**Comment by chess99** [comment]
ok, I will cancel the PR.
modifies in this PR work fine on my computer ( using conda 3.8.8 ). in fact I'm not very familiar with Python, help needed~

two problems I encountered when I run visualmetrics.py :
1. deadlock on proc.poll( )
2. TypeError: cannot use a string pattern on a bytes-like object

**Comment by chess99** [comment]
> It would be better to pass universal_newlines=True to the Popen instead. That way it will be text output on both Pythoin 2.7 and 3.x. Otherwise, I think the decode might fail on Python 2.7 (which the code still supports).

PR updated

File: visualmetrics.py [hunk 1]
```diff
@@ -141,9 +141,9 @@ def extract_frames(video, directory, full_resolution, viewport):
                    os.path.join(dir_escaped, 'img-%d.png')]
         logging.debug(' '.join(command))
         lines = []
-        proc = subprocess.Popen(command, stderr=subprocess.PIPE)
+        proc = subprocess.Popen(command, stderr=subprocess.PIPE, universal_newlines=True)
         while proc.poll() is None:
-            lines.extend(iter(proc.stderr.readline, ""))
+            lines.append(proc.stderr.readline())
```
Comment (pmeenan) [inline visualmetrics.py:146]: Can you revert this change as well? The existing code reads all of the pending lines in one pass.

Comment (chess99) [reply visualmetrics.py:146]: I encountered deadlock when program reach here

conda 4.10.1
python 3.8.8
